### PR TITLE
distributor: conditionally export replication factor; export ingest storage metric

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
The distributor currently exports `cortex_distributor_replication_factor` regardless whether ingest storage is enabled or not. This metric is deceiving when using ingest storage because it doesn't affect anything.

This PR exports `cortex_distributor_replication_factor` only when ingest storage is disabled. Additionally, the distributor now exports a single constant metric `cortex_distributor_ingest_storage_enabled` when ingest storage is enabled.

Some recording rules and dashboards rely on `cortex_distributor_replication_factor` and render incorrect results with ingest storage. I plan to correct them in follow-up PRs.

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/distributor/distributor.go [hunk 1]
```diff
@@ -482,6 +476,20 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 		subservices = append(subservices, d.ingestStorageWriter)
 	}
 
+	// Register each metric only if the corresponding storage is enabled.
+	// Some queries in the mixin use the presence of these metrics as indication whether Mimir is running with ingest storage or not.
+	if cfg.IngestStorageConfig.Enabled {
+		promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "cortex_distributor_ingest_storage_enabled",
+			Help: "Whether writes are being processed via ingest storage. Always 1 if ingest storage is enabled.",
+		}).Set(1)
```
Comment (pstibrany) [inline pkg/distributor/distributor.go:485]: Shouldn't we always export this one, and set value to 0, if it's not enabled?

File: pkg/distributor/distributor.go [hunk 2]
```diff
@@ -482,6 +476,20 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
...
+	} else {
+		promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "cortex_distributor_replication_factor",
+			Help: "The configured replication factor.",
+		}).Set(float64(ingestersRing.ReplicationFactor()))
```
Comment (pstibrany) [inline pkg/distributor/distributor.go:490]: I think this deserves a `[CHANGE]` in the changelog.

Comment (dimitarvdimitrov) [reply pkg/distributor/distributor.go:490]: we haven't entered any of the sigyn changes in the changelog, so i wasn't sure what to do here. I can add one

Comment (pstibrany) [reply pkg/distributor/distributor.go:490]: Oh, that's a good point. This only applies to ingest-storage. We should gather all changelog entries, and not forget about this once we write them.

File: pkg/distributor/distributor.go (hunk 1 above)
Comment (dimitarvdimitrov) [reply pkg/distributor/distributor.go]: did it in https://github.com/grafana/mimir/pull/7809/commits/1b734cf71e8fbc03d31e73e6f1a8b477d49e3d8f

File: pkg/distributor/distributor.go (hunk 2 above)
Comment (dimitarvdimitrov) [reply pkg/distributor/distributor.go]: added in https://github.com/grafana/mimir/pull/7809/commits/1b734cf71e8fbc03d31e73e6f1a8b477d49e3d8f

**Review by pstibrany — APPROVED** [review]

File: pkg/distributor/distributor.go (hunk 2 above)
Comment (dimitarvdimitrov) [reply pkg/distributor/distributor.go]: in this case i'll remove the entry. GH didn't show your comment when i left mine

**Pull request merged**